luda-editor/luda-editor-client/src/app/sequence_list/list_item.rs

```diff
@@ -0,0 +1,84 @@
+use super::{
+    events::SequenceListEvent,
+    rounded_rectangle::RoundedRectangleColor,
+    types::{RenderingTreeRow, RenderingTreeRows, SequenceLoadStateDetail},
+    SequenceList, BUTTON_HEIGHT, MARGIN,
+};
+use crate::app::sequence_list::SPACING;
+use namui::{render, Wh};
+
+impl SequenceList {
+    pub fn render_list_item(&self, width: f32, title: &String) -> RenderingTreeRow {
+        let path = format!("sequence/{}", title);
+        let element_width = width - 2.0 * MARGIN;
+        let button_wh = Wh {
+            width: element_width,
+            height: BUTTON_HEIGHT,
+        };
+        let title_load_state = self.sequence_load_state_map.get(&path);
+        let is_open = title_load_state.is_some();
+        let mut elements: Vec<RenderingTreeRow> = Vec::new();
+
+        elements.push(RenderingTreeRow::new(
+            render![
+                self.render_rounded_rectangle(button_wh, RoundedRectangleColor::DarkGray)
+                    .with_mouse_cursor(namui::MouseCursor::Pointer)
+                    .attach_event(move |builder| {
+                        let path = path.clone();
+                        builder.on_mouse_down(move |_| match is_open {
+                            true => {
+                                let path = path.clone();
+                                namui::event::send(
+                                    SequenceListEvent::SequenceLoadStateUpdateEvent {
```

Comment:
Personally, I found it very difficult to read the code here. it maybe because of the name of event or variables.

`if is open` <- has no subject. So when I read code, I thought about 'what can be opened'. 'opening file'? 'opening... what?'

I really found what I can open when I run the demo. AH. The list inside of button! HOW CAN I KNOW THAT only reading this code :S

the code must be nicer if it like this,
```
if list(or inner_list) is open {
  close it()
} else {
  open it()
}
fn close {
  // put what I have to do to close
}
fn open {
  // put what I have to do to open
  start load()
}
```

It's very important to divide code into functions because when we read the code, it's not our business what exactly your code do.
It's important how the logic is flowing. That mean the code should hide the implementation, just show the interaction between interfaces, functions with good name to feel the flow.
After read all flow, I read the implementation. It's lower priority. logic is more important than implementation, that mean we have to write the code to let programmer know how the logic is.

So anyway, would you like to fix this code more readable?

Reply:
It's good if the code inside doesn't know the outside. Problems arise the moment you know the outside from the inside.